src/main/java/frc/robot/commands/DriveWithJoysticks.java

```diff
@@ -66,19 +78,45 @@ public void execute() {
         joystickInputsFiltered[0] = Math.pow(joystickInputsFiltered[0], 1);
         joystickInputsFiltered[1] = Math.pow(joystickInputsFiltered[1], 1);
 
-        xMpS = joystickInputsFiltered[0] * DriveConstants.MaxSpeedMetPerSec;
-        yMpS = joystickInputsFiltered[1] * DriveConstants.MaxSpeedMetPerSec;
-        rotRadpS = Deadband.oneAxisDeadband(rot.getAsDouble(), DriveConstants.deadbandPercent);
-        rotRadpS = Math.pow(rotRadpS, 1) * DriveConstants.MaxAngularRateRadiansPerSec;
+        commandedXMpS = joystickInputsFiltered[0] * DriveConstants.MaxSpeedMetPerSec;
+        commandedYMpS = joystickInputsFiltered[1] * DriveConstants.MaxSpeedMetPerSec;
+        commandedRotRadpS =
+                Deadband.oneAxisDeadband(rot.getAsDouble(), DriveConstants.deadbandPercent);
+        commandedRotRadpS =
+                Math.pow(commandedRotRadpS, 1) * DriveConstants.MaxAngularRateRadiansPerSec;
 
         if (babyMode.getAsBoolean()) {
-            xMpS *= 0.5;
-            yMpS *= 0.5;
-            rotRadpS *= 0.5;
+            commandedXMpS *= 0.5;
+            commandedYMpS *= 0.5;
+            commandedRotRadpS *= 0.5;
+        }
+
+        currentXMpS = currentVelocitySupplier.get().get(0, 0);
+        currentYMpS = currentVelocitySupplier.get().get(1, 0);
+
+        double currentTime = Utils.getCurrentTimeSeconds();
+        double diffTime = currentTime - lastTime;
+        lastTime = currentTime;
+
+        if (Math.hypot(currentXMpS, currentYMpS) < Math.hypot(commandedXMpS, commandedYMpS)) {
+            double velocityVectorTheta = Math.atan2(commandedYMpS, commandedXMpS);
+            commandedXMpS =
+                    currentXMpS
+                            + Math.signum(commandedXMpS)
+                                    * (DriveConstants.maxAccelerationMetersPerSecSquared
+                                            * Math.cos(velocityVectorTheta)
+                                            * diffTime);
+            commandedYMpS =
+                    currentYMpS
+                            + Math.signum(commandedYMpS)
+                                    * (DriveConstants.maxAccelerationMetersPerSecSquared
+                                            * Math.sin(velocityVectorTheta)
+                                            * diffTime);
         }
 
         drivetrain.setGoalChassisSpeeds(
-                new ChassisSpeeds(xMpS, yMpS, rotRadpS), fieldCentric.getAsBoolean());
+                new ChassisSpeeds(commandedXMpS, commandedYMpS, commandedRotRadpS),
+                fieldCentric.getAsBoolean());
```

Comment:
For memory related reasons: could we maintain a ChassisSpeeds private variable and update it rather than creating a new object every cycle?